### DISCUSSION TITLE
Neovim is changing to use the upstream treesitter highlighting implementation (as used by Helix)

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#6487] [General] [answered]
https://github.com/neovim/neovim/issues/22495
https://github.com/neovim/neovim/pull/22518 (merged)

Hopefully this will be a big help to Helix enabling everyone to share the same highlighting queries. Can anyone with a better understanding comment?

**Comment by archseer** [comment, marked as answer]
I can't find the original thread but we're working on aligning with them on some of the queries so there could be more reuse/queries directly provided by upstream

**Reply by the-mikedavis** [reply]
Most of the discussion is in https://github.com/nvim-treesitter/nvim-treesitter/issues/3944 IIRC
